### PR TITLE
QA-161 add new exceptions of infrastructure and bundles

### PR DESCRIPTION
**Pull request by skhomuti**

### CODE REVIEW

File: src/adcm_pytest_plugin/exceptions/bundles.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    @classmethod
+    def raise_if_suitable(cls, message):
+        """Raise suitable bundle error if possible"""
+        __tracebackhide__ = True  # pylint: disable=unused-variable
+
+        if "Wrong context" in message or "AnsibleFilterError" in message:
+            raise AnsibleError(message)
+        if "control process exited with error" in message:
+            raise SystemctlError(message)
+        if "Cannot allocate memory" in message:
```
Comment (vgorkavenko) [inline src/adcm_pytest_plugin/exceptions/bundles.py:30]: Can we raise a warning if we can't match msg and error class?

Comment (skhomuti) [reply src/adcm_pytest_plugin/exceptions/bundles.py:30]: I'm not sure what this is needed. Not all errors we can categorize and it's ok

File: src/adcm_pytest_plugin/steps/common.py [hunk 2]
```diff
@@ -38,24 +39,26 @@ def assume_step(title, exception=None):
         raise ValueError("This is expected exception")
 
     """
-    if callable(title):
+    if callable(title):  # pylint: disable=no-else-return
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/steps/common.py:42]: I can clearly see a return on else clause here. Could you clarify why this mute is here?

File: src/adcm_pytest_plugin/steps/common.py [hunk 3]
```diff
@@ -38,24 +39,26 @@ def assume_step(title, exception=None):
...
         return AssumeStepContext(title.__name__, exception)(title)
     else:
         return AssumeStepContext(title, exception)
 
 
 class AssumeStepContext:
+    """Step wrapper like `allure.step`"""
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/steps/common.py:49]: I think this docstring should be more detailed

File: src/adcm_pytest_plugin/steps/common.py (hunk 2 above)
Comment (skhomuti) [reply src/adcm_pytest_plugin/steps/common.py:42]: similar fix https://github.com/arenadata/adcm_pytest_plugin/pull/54/files#diff-c7d550aaf609be6873c469b90e640e7da283e1e2db0bf54736f5b92109c689e6R182-R186
it's a strange inspection and I don't really want to fix it here

Comment (skhomuti) [reply src/adcm_pytest_plugin/steps/common.py]: fix

File: src/adcm_pytest_plugin/steps/common.py (hunk 3 above)
Comment (skhomuti) [reply src/adcm_pytest_plugin/steps/common.py]: fix

**Review by dgusakov — APPROVED** [review]

**Review by vgorkavenko — APPROVED** [review]

**Pull request merged**